k4MarlinWrapper/src/components/MarlinProcessorWrapper.cpp

```diff
@@ -249,7 +249,10 @@ StatusCode MarlinProcessorWrapper::execute() {
     } else {
       m_processor->processEvent(the_event);
     }
+
+    m_processor->setFirstEvent(false, this);
```

Comment:
This isn't thread-safe though.

Reply:
Especially as this comes after processEvent, so this means that not only are things not reproducible because of the randomseed, but sequential is the only way to go then.

Reply:
Yes, that is true. Didn't we agree on "sequential only" for now for all wrapped processors? I am not sure how many would actually properly cope with being in run on multiple threads. How does Gaudi do things in this case?

I think the more important bit here is to get things to run as expected sequentially, otherwise the whole "run processors unchanged" does not really hold, IMHO.

Reply:
Instead of "firstEvent" things like that are done in initialise, holds true for Gaudi, and also most marlin processors.

So far I think we said: thinks can run in parallel, but not in a reproducible way. So this "isFirstEvent" is mostly something we need to document as limitiations.
https://github.com/key4hep/k4MarlinWrapper/blob/master/doc/starterkit/k4MarlinWrapperCLIC/howtoMultithread.md

Reply:
Yeah, initializing these things before the event loop is how it should be done. [A quick query](https://github.com/search?q=org%3AiLCSoft+isFirstEvent&type=code) at least shows that there are a few existing processors that rely on `isFirstEvent` to work to register some histogram.

I will add this limitation to the documentation.

Reply:
Thanks!